### DISCUSSION TITLE
More detailed HTTP status codes?

### DISCUSSION THREAD

**Discussion by GD-Dal** [extremenetworks/ExtremeCloudIQ-APIs#10] [Q&A] [answered]
Hi
I'm writing a few python scripts to add / update endusers i Extremecloud IQ via python / API.
But it is trial and error, but the only "help" I'm given when something is wrong is:
```
Error adding PPSK user - HTTP Status Code: 400
<Response [400]>
This is given by print (response.status_code)
```
Is there a way to get more detailed error messages, or does it simply not exist?

So far I have been given Status Code 400 (and nothing else) when:
- username already exists
- usergroup is not assigned to any SSID
- using the usergroup id in the post request (when adding user)
- password is not within given parameters of the user group

Thanks.

**Comment by timjsmith24** [comment]
The error messages are returned in json format.

**Comment by timjsmith24** [comment, marked as answer]
Try decoding the response then look for the error message. Something like this.

try:
    data = response.json()
except json.JSONDecodeError:
    print("Unable to parse the data from json, script cannot proceed")
if 'error' in response:
    if response['error_mssage']:
        print(f"Status Code {response['error_id']}: {response['error_message']}")

**Comment by GD-Dal** [comment]
Excellent.
Thank you
